### ISSUE TITLE
WorkingDir of subdir of a volume mount is not found by Python's os.getcwd()

### ISSUE THREAD

**Issue by adamkerz** [docker/for-win#5698]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: BCFAE496-38CD-4C65-874D-E51FB949BA6B/20200205045509

### Expected behavior

Running `python -c "import os; os.getcwd()"` should succeed

### Actual behavior

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory
```

Same problem when running `os.getcwd()` from within a script.

### Information

I am running docker desktop 2.2.0.0 (42247), engine 19.03.5
It is not present in docker desktop 2.1.0.3, engine 19.03.2
Windows: 1909 (18363.628), not virtualised

### Steps to reproduce the behavior

Dockerfile
```
FROM python:3.6.8-stretch
WORKDIR /my_dir/working
```

Run in cmd:
```
set CWD_BUG_DIR=C:\cwd_1_bug\my_dir
mkdir %CWD_BUG_DIR%
mkdir %CWD_BUG_DIR%\working

docker build --no-cache -t docker_cwd_bug .
# bug
docker run -it -v %CWD_BUG_DIR%:/my_dir docker_cwd_bug python -c "import os; os.getcwd()"
# ok - mounting the actual working dir appears to avoid the issue
docker run -it -v %CWD_BUG_DIR%\working:/my_dir/working docker_cwd_bug python -c "import os; os.getcwd()"
```

**Comment by mat007** [comment]
I think this might be a variant of #5550 ?

**Comment by mat007** [comment]
I wasn’t able to reproduce it by the way, does it occur every time for you?
Also I’m running this new private which fixes some related issues, would you mind trying it as well?
It’s https://download-stage.docker.com/win/stable/42579/Docker%20Desktop%20Installer.exe

**Comment by adamkerz** [comment]
Yep, every time. OK, I'll have a go of the private version tomorrow.

**Comment by vchirikov** [comment]
@mat007 thanks, the issue with fs is fixed.

**Comment by AlJohri** [comment]
I briefly experienced this same error in Docker for Mac 2.2.0.0. I discovered it when random things with Python's pathlib were failing such as running `.resolve()` or `.stem`. The root cause was that python's `os.getcwd()` was returning file not found error. I'm using vscode with a docker container which mounts the current file system directory.

Closing vscode and starting again somehow resolved the error for now?
